enhancements/hypershift/performanceprofilecontroller-node-tuning.md

```diff
@@ -0,0 +1,328 @@
+      "hypershift.openshift.io/nodePool" : NodePool.Name
+    ]
+    annotations: [
+      "hypershift.openshift.io/nodePool" : NodePoolNamespacedName
+    ]
+  }
+```
+### API Extensions
+
+None.
+
+This proposal will use the API extensions already proposed by the NTO adaptation [see](/enhancements/hypershift/node-tuning.md#api-extensions)
+
+### Risks and Mitigations
+
+N/A.
+
+### Drawbacks
+
+N/A.
+
+### Test Plan
+
+N/A.
+
+#### Dev Preview -> Tech Preview
+
+N/A.
+
+#### Tech Preview -> GA
+
+N/A.
+
+#### Removing a deprecated feature
+
+N/A.
+
+### Upgrade / Downgrade Strategy
+
+### Version Skew Strategy
+
+N/A.
+
+### Operational Aspects of API Extensions
+
+N/A.
+
+#### Failure Modes
+
+#### Support Procedures
+
+## Alternatives
+
+## Design Details
+
+### Hypershift NodePool controller
+In order to propagate any `PerformanceProfile` inside a configmap referenced in the `.spec.TunedConfig` NodePool field to the hosted control plane namespace where the Performance Profile controller could read it, NodePool controller should be modified.
+Tuned manifests are already being handled in NodePool controller so we would apply minimal changes to that procedure to also use it for PerformanceProfile manifests.
+
+Basically all configmaps referenced in `NodePool.spec.TunedConfig` field will be read and all `PerformanceProfile`s found embedded into them will be extracted, appended and embedded in a new configmap, this time in the hosted control plane namespace, where Performance Profile controller will be able to handle them.
+See openshift/hypershift#1782 
+
+To be able to handle `KubeletConfig` propagation properly NodePool controller should be changed to recognize a `KubeletConfig` as a valid content and set its defaults properly, that is mainly to ensure label `machineconfiguration.openshift.io/role=worker` is set.
+This can be made by changing [`defaultAndValidateConfigManifest`](https://github.com/openshift/hypershift/blob/fa0ca3d09fab02ebff64d45b97cc1abaf4f1c27a/hypershift-operator/controllers/nodepool/nodepool_controller.go#L1439)  to handle `KubeletConfig` almost as it is handling `MachineConfig` right now.
+
+### Performance Profile controller
+
+1. Change SetUp to watch over ConfigMaps with Label: `hypershift.openshift.io/performanceprofile-config=true`
+In Hypershift Performance Profile controller (PPC) will not be reconciling PerformanceProfile objects perse but ConfigMap objects with PerformanceProfile objects embedded, so the controller SetUp should be changed to resemble this.
+
+2. Change reconcile loop
+  Reconcile should somehow follow the same operations as in StandAlone but with some changes
+   - Extract PerformanceProfile object from ConfigMap `config` field.
+   - Handle deletion
+     - If ConfigMap has `DeletionTimestamp != nil` should handle deletion of the objects created by the PerformanceProfile inside de ConfigMap.
+       - **TODO** Do not know how to do this as these elements are in the hosted cluster.
```

Comment:
Do you have a way of identifying which configmaps containing Tuned,MachineConfig,KubeletConfig resources were generated for a particular PerformanceProfile so you know what to delete in case the PerformanceProfile is deleted?

Reply:
Yeah all the ConfigMaps generated by a PerformanceProfile has a label with the PerformanceProfile.Name that originated them. It is in the diagram ( maybe I should also document it somewhere else ) 

Apart from that I have to update this paragraph. Right now I have added a "finalizer" to any Configmap with a PerformanceProfile embedded, to be able to handle deletion but  just delete RtClass, as right now all the other components created from this PP are embedded into ConfigMaps which has an OwnerReference with the PP configmap and will be deleted by k8s machinery triggering the deletion procedure of the embedded elements.